msdsales/read.go

```diff
@@ -0,0 +1,95 @@
+package msdsales
+
+import (
+	"context"
+	"fmt"
+	"math"
+	"net/url"
+	"strings"
+
+	"github.com/amp-labs/connectors/common"
+)
+
+// TODO MS Sales allows fine control of reading
+// TODO basic read
+// $select = ReadParams{fields} [COMPLETED]
+// $expand = nested response
+// $orderby = list of fields with asc/desc keyword
+// TODO batch
+// $apply = batch
+// TODO search
+// $filter = query functions, comparisons
+// TODO pagination
+// $top = <int> of entries to return (ignored if header <Prefer: odata.maxpagesize>)
+// $count = counts all existing rows (@odata.count)
```

Comment:
Microsoft's Read is very flexible, see [options](https://learn.microsoft.com/en-us/power-apps/developer/data-platform/webapi/query-data-web-api#odata-query-options). Should we support all query capabilities?

Reply:
https://www.loom.com/share/5b109408daf9416aa07ac9427e13e228

Reply:
https://www.loom.com/share/5b109408daf9416aa07ac9427e13e228

Reply:
Yes, bottom line is I have described these options in docs.